sys/include/net/sock/dtls.h

```diff
@@ -547,25 +567,18 @@ int sock_dtls_create(sock_dtls_t *sock, sock_udp_t *udp_sock,
                      credman_tag_t tag, unsigned version, unsigned role);
 
 /**
- * @brief Creates a new DTLS session
- *
- * Initializes handshake process with a DTLS server  at @p ep.
+ * @brief Send a ClientHello to start the handshake
  *
- * @param[in]  sock     DLTS sock to use
- * @param[in]  ep       Remote endpoint of the session
- * @param[out] remote   The created session, cannot be NULL
+ * @param[in] sock      DTLS sock to use
+ * @param[in] ep        Remote endpoint to start a handshake with
+ * @param[out] remote   Resulting session
  *
- * @return  0 on success
- * @return  -EAGAIN, if DTLS_HANDSHAKE_TIMEOUT is `0` and no data is available.
- * @return  -EADDRNOTAVAIL, if the local endpoint of @p sock is not set.
- * @return  -EINVAL, if @p remote is invalid or @p sock is not properly
- *          initialized (or closed while sock_udp_recv() blocks).
```

Comment:
Shouldn't those two still be relevant for the new behavior?

Reply:
Should we leave it there even though we don't have a way to check it? For tinydtls, `dtls_connect` only returns either 1, 0, or -1, not much information on what causes it.

Reply:
Other DTLS libraries might still, so I would say, yes!

Reply:
(you also could check yourself, for both `EINVAL` and `EADDRUNAVAIL`, before calling `dtls_connect` ;-))

Reply:
readded `EINVAL` and `EADDRUNAVAIL`

Reply:
Only to `_create()`, not `_init()` ;-)

Reply:
:facepalm: should be added to both now.